packages/framer-motion/src/animation/sequence/create.ts

```diff
@@ -330,8 +332,11 @@ export function getValueTransition(
     transition: DynamicAnimationOptions & At,
     key: string
 ): DynamicAnimationOptions {
-    return transition[key]
-        ? { ...transition, ...transition[key] }
+    return transition[key as keyof typeof transition]
```

Comment:
@mattgperry how about depending on:
- return type
- and global per-repo type override for loops, so the key is auto-inferred from the array
  - you should also consider to switch to Object.keys for a slight performance boost 
  - https://dev.to/tmhao2005/typescript-override-typing-for-objectkeys-5dmh

It would ensure cases like this don’t need manual `as` override, removing possibilities of human errors.

If you still want to manually define `key`’s type, then you should do it [once before the for loop](https://stackoverflow.com/a/59656330).